core/src/main/java/org/axonframework/eventsourcing/EventSourcingRepository.java

```diff
@@ -240,6 +240,31 @@ public EventSourcingRepository(AggregateModel<T> aggregateModel, AggregateFactor
         this.repositoryProvider = repositoryProvider;
     }
 
+    /**
+     * Initializes a repository with a locking strategy underpinned by the given {@code lockFactory}, uses the given
+     * {@code aggregateFactory} to create new aggregate instances and triggering snapshots using the given
+     * {@code snapshotTriggerDefinition}
+     *
+     * @param aggregateModel            The meta model describing the aggregate's structure
+     * @param lockFactory               The lock
```

Comment:
I'd rephrase the javadoc for `lockFactory` to `The locking strategy to apply to this repository` to be inline with the other constructors their javadoc where the `LockFactory` is included.